src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,42 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Locale;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern(
+            "dd.MM.yyyy", Locale.ENGLISH);
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        int[] salaryForPeriod = new int[names.length];
+        LocalDate beginDate = LocalDate.parse(dateFrom, DATE_TIME_FORMATTER);
+        LocalDate endDate = LocalDate.parse(dateTo, DATE_TIME_FORMATTER);
+        for (int i = 0; i < names.length; i++) {
+            int employerSalary = 0;
+            for (String datum : data) {
+                String[] dataFormatted = datum.split(" ");
+                LocalDate workingDayDate = LocalDate.parse(dataFormatted[0], DATE_TIME_FORMATTER);
+                if (workingDayDate.isAfter(beginDate.minusDays(1))
+                        && workingDayDate.isBefore(endDate.plusDays(1))
```

Comment:
We may avoid using minusDays and plusDays methods
```suggestion
                if (!workingDayDate.isAfter(endDate)
                        && !workingDayDate.isBefore(beginDate)
```